modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/DelegatedStakesRoutes.scala

```diff
@@ -0,0 +1,121 @@
+package io.constellationnetwork.dag.l0.http.routes
+
+import cats.data.Validated.{Invalid, Valid}
+import cats.effect.Async
+import cats.syntax.all._
+
+import io.constellationnetwork.dag.l0.domain.delegatedStake.{CreateDelegatedStakeOutput, DelegatedStakeOutput, WithdrawDelegatedStakeOutput}
+import io.constellationnetwork.ext.http4s.AddressVar
+import io.constellationnetwork.kernel._
+import io.constellationnetwork.node.shared.domain.delegatedStake.UpdateDelegatedStakeValidator
+import io.constellationnetwork.node.shared.domain.node.NodeStorage
+import io.constellationnetwork.node.shared.domain.snapshot.storage.SnapshotStorage
+import io.constellationnetwork.node.shared.infrastructure.snapshot.storage.SnapshotLocalFileSystemStorage
+import io.constellationnetwork.routes.internal._
+import io.constellationnetwork.schema._
+import io.constellationnetwork.schema.address.Address
+import io.constellationnetwork.schema.delegatedStake.UpdateDelegatedStake
+import io.constellationnetwork.schema.peer.PeerId
+import io.constellationnetwork.security.signature.Signed
+
+import derevo.cats.{eqv, show}
+import derevo.circe.magnolia.encoder
+import derevo.derive
+import eu.timepit.refined.auto._
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityCodec.circeEntityEncoder
+import org.http4s.dsl.Http4sDsl
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+final case class DelegatedStakesRoutes[F[_]: Async](
+  mkCell: DelegatedStakeOutput => Cell[F, StackF, _, Either[CellError, Ω], _],
+  validator: UpdateDelegatedStakeValidator[F],
+  snapshotStorage: SnapshotStorage[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo],
+  fullGlobalSnapshotStorage: SnapshotLocalFileSystemStorage[F, GlobalSnapshot],
+  nodeStorage: NodeStorage[F]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  import org.http4s.circe.CirceEntityCodec.circeEntityDecoder
+
+  private val logger = Slf4jLogger.getLoggerFromName[F]("DelegatedStakesLogger")
+
+  protected val prefixPath: InternalUrlPrefix = "/"
+
+  private def readLatestFullSnapshot(maybeOrdinal: Option[SnapshotOrdinal]): F[Option[Signed[GlobalSnapshot]]] =
+    maybeOrdinal.traverse(fullGlobalSnapshotStorage.read).map(_.flatten)
+
+  private def getLatestFullSnapshot: F[Option[Signed[GlobalSnapshot]]] =
+    for {
+      maybeOrdinal <- snapshotStorage.headSnapshot.map(_.map(_.ordinal))
+      maybeSnapshot <- readLatestFullSnapshot(maybeOrdinal)
+    } yield maybeSnapshot
+
+  @derive(eqv, show, encoder)
+  case class DelegatedStakesInfo(
```

Comment:
It might make more sense to locate this alongside your other schemas (not directly embedded in the class,) in case it is ever used elsewhere, I realize that's unlikely for now, but it's easier for someone trying to find all the schemas related to this feature.

Additionally, for all the input / output types, please implement the model yml, here's an example api doc update PR for reference which includes new type definitions: https://github.com/Constellation-Labs/tessellation/pull/1034